### PR TITLE
Test suite fixes

### PR DESCRIPTION
**Pull request by peterbarker**
- check to see if we need to reboot after every test, not just exceptions (and cope with fallout from that)
 - remove pointless code
 - remove pointless context

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -512,10 +512,6 @@ def fly_battery_failsafe(self, timeout=300):
         # wait for LAND mode. If unsuccessful an exception will be raised
         self.wait_mode('LAND', timeout=timeout)
 
-        # disable battery failsafe
-        self.set_parameter('BATT_FS_LOW_ACT', 0)
-        self.set_parameter('SIM_BATT_VOLTAGE', 13)
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: If you are removing this, shouldn't there be a context_push/pop here?

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -616,20 +612,9 @@ def fly_fence_test(self, timeout=180):
...
-                self.mavproxy.send('arm uncheck all\n')
-                self.arm_vehicle()
-                # remove if we ever clear battery failsafe flag on disarm:
-                self.mavproxy.send('arm check all\n')
-                self.progress("Reached home OK")
                 return
 
-        # disable fence, enable avoidance
-        self.set_parameter("FENCE_ENABLE", 0)
-        self.set_parameter("AVOID_ENABLE", 1)
```
Comment (OXINARF) [inline Tools/autotest/arducopter.py]: Same.

**Comment by khancyr** [comment]
@OXINARF context_pop() is now done at the end of each test by the run_one_test() function, that why there are those removal as they are duplicate of context_pop()

**Comment by OXINARF** [comment]
@khancyr Ah! Thanks, I wasn't aware of that.

**Pull request closed without merging**